File: 2021/day1.go
package main

import ("fmt"; "io/ioutil"; "log"; "strings"; "strconv")

func handle(err error) {
	if err != nil {
		log.Fatal(err)
	}
}
func Int(str string) int {
	ret, err := strconv.Atoi(str)	
	handle(err)
	return ret
}
func part1(filename string) int {
	content, err := ioutil.ReadFile(filename)
	handle(err)
	lines := strings.Split(string(content), "\n")
	increasing := 0
	prev := Int(lines[0])
	for _, line := range lines[1:] {
		curr := Int(line) 
		if curr > prev {
			increasing += 1
		}
		prev = curr
	}
	return increasing
}

func part2(filename string) int {
	content, err := ioutil.ReadFile(filename)
	handle(err)
	lines := strings.Split(string(content), "\n")
	increasing := -1 
	prev := 0
	for i, _ := range lines[2:] {
		a := Int(lines[i])
		b := Int(lines[i+1])
		c := Int(lines[i+2])
		curr := a + b + c 
		if curr > prev {
			increasing += 1
		}
		prev = curr
	}
	return increasing
}


func main() {
	fmt.Printf("Part 1: %d\n", part1("inputday1.txt"))
	fmt.Printf("Part 2: %d\n", part2("inputday1.txt"))
}

File: 2022/day1.js
const fs = require('fs')
const fn = 'inputday1.txt'

fs.readFile(fn, 'utf8', (err, data) => {
    if (err) {
        console.error(err)
        return
    }

    // Part 1  
    const lines = data.split('\n')
    part1 = 0
    temp = 0
    for (let i = 0; i < lines.length; i++) {
        if (lines[i] != '') {
            temp += parseInt(lines[i])
        }
        else {
            if (temp > part1) {
                part1 = temp
            }
            temp = 0
        }
    }
    console.log(`Part 1: ${part1}`)

    // Part 2
    calories = []
    for (let i = 0; i < lines.length; i++) {
        if (lines[i] != '') {
            temp += parseInt(lines[i])
        }
        else {
            calories.push(temp)
            temp = 0
        }
    }
    calories.sort((a, b) => b - a)
    part2 = calories[0] + calories[1] + calories[2]
    console.log(`Part 2: ${part2}`)
})